sympy/functions/elementary/trigonometric.py

```diff
@@ -814,6 +839,31 @@ def eval(cls, arg):
                 return None
 
             if pi_coeff.is_Rational:
+                if not pi_coeff.q % 2:
+                    narg = pi_coeff*S.Pi*2
+                    cresult, sresult = cos(narg), cos(narg - S.Pi/2)
+                    if not isinstance(cresult, cos) \
+                            and not isinstance(sresult, cos):
+                        if sresult == 0:
+                            return S.ComplexInfinity
```

Comment:
Could you add a test that covers this line? While you are editing, could you also add space around the `-` in the next line (and in other places that I mark with "pep8")?